consul/autopilot.go

```diff
@@ -68,11 +84,26 @@ func (s *Server) autopilotLoop() {
 	}
 }
 
+// lowestServerVersion returns the lowest version among the alive servers
```

Comment:
I'd factor this out into a companion to https://github.com/hashicorp/consul/blob/master/consul/util.go#L65-L92, like `AreServersAtLeastOnVersion`. Or, since have util depend on consul/server might be weird you could move them both into a new Go file in the Consul package like "compatibility.go" so they live side-by-side.